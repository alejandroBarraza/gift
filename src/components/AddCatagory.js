import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCatagory = ({ setcategories }) => {
    const [inputValue, setinputValue] = useState('');

    //function update input text
    const handleText = ({ target: { value } }) => {
        setinputValue(value);
    };

    //function update input value in the list
    const handleSubmit = (e) => {
        e.preventDefault();
        setcategories((cat) => [inputValue,...cat]);
        setinputValue('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <h2>Add a new category</h2>
            <input type='text' required value={inputValue} onChange={handleText} />
        </form>
    );
};

AddCatagory.propTypes = {
    setcategories: PropTypes.func.isRequired,
};
