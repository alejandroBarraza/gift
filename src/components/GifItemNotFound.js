import React from 'react'

export const GifItemNotFound = ({category}) => {
    return (
        <div>
            <h2>{category} no encontrada, </h2>
            
        </div>
    )
}
