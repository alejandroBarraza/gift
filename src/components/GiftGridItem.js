import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({ url, title }) => {
    return (
        <div className='card glassmorphismo'>
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    );
};

GiftGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};
