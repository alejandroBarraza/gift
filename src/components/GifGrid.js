import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

import { GiftGridItem } from './GiftGridItem';

export const GifGrid = ({ category }) => {
    const { data: imgs } = useFetchGifs(category);
    console.log(imgs);
    console.log(...imgs);

    return (
        <>
            <h2> {category}</h2>

            <div className='card-container'>
                {imgs.map((img) => (
                    <GiftGridItem key={img.id} {...img} />
                ))}
            </div>
        </>
    );
};
