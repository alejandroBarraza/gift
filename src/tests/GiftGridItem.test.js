import '@wojtekmaj/enzyme-adapter-react-17';
import { GiftGridItem } from '../components/GiftGridItem';
import '../setupTests';
import { shallow } from 'enzyme';

describe('testing gridGridItem in enzyme', () => {
    let wrapper;
    const url = 'https://localhost:3000';
    const title = 'naruto';

    beforeEach(() => {
        wrapper = shallow(<GiftGridItem url={url} title={title} />);
    });
    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
