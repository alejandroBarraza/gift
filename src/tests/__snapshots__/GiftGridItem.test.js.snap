// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing gridGridItem in enzyme should match snapshot 1`] = `
<div
  className="card glassmorphismo"
>
  <p>
    naruto
  </p>
  <img
    alt="naruto"
    src="https://localhost:3000"
  />
</div>
`;
